### PR TITLE
NetworkPkg\Ip6Dxe and MdeModulePkg\Universal\SetupBrowserDxe: Fix Previous CodeQL fix

### PR DESCRIPTION
**Pull request by pohanch**
# Preface

Fix potential errors from CodeQL fixes ([commit a9342ca](https://github.com/microsoft/mu_basecore/commit/a9342ca89e8a5e6eb4f7ee7649af2cb22403d763)) and ([commit 3657711](https://github.com/microsoft/mu_basecore/commit/3657711bd5d8c26adaea3db9931afecd517fca7f)) that resulted in an uninitialized variable attempting to be used.

## Description

BufferSize should not be used when StrPtr is NULL, so wrap BufferSize section with if statement.
RetVal may not be initialized if the logic does not route through an if statement.
Status may not be initialized if NeighborCache is NULL. Set Status to EFI_NOT_FOUND.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Built several times with the change locally, and the system boots.

## Integration Instructions

N/A

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

File: MdeModulePkg/Universal/SetupBrowserDxe/Setup.c [hunk 1]
```diff
@@ -3167,7 +3167,7 @@ ConfirmSaveFail (
 {
   CHAR16  *FormTitle;
   CHAR16  *StringBuffer;
-  UINT32  RetVal;
+  UINT32  RetVal = BROWSER_ACTION_UNREGISTER; // MU_CHANGE - Initialize variable that might not be updated due to error checking
```
Comment (spbrogan) [inline MdeModulePkg/Universal/SetupBrowserDxe/Setup.c]: coding convention requires init independent of declaration.  I know we don't always follow this so you will see code that isn't compliant but for new changes we should. 

UINT32  RetVal;

RetVal = BROWSER_ACTION_UNREGISTER;

**Review by spbrogan — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Comment by pohanch** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

**Comment by pohanch** [comment]
> @pohanch please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree company="Microsoft"
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Comment by pohanch** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: MdeModulePkg/Universal/SetupBrowserDxe/Setup.c (hunk 1 above)
Comment (pohanch) [reply MdeModulePkg/Universal/SetupBrowserDxe/Setup.c]: Separated initialization and declaration.

**Pull request merged**